pkg/kubelet/lifecycle/handlers.go

```diff
@@ -128,8 +171,30 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	path := handler.HTTPGet.Path
+	scheme := "http"
+
+	url := formatURL(scheme, host, port, path)
+	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
+	if err != nil {
+		return "", err
+	}
+	header := buildHeader(handler.HTTPGet.HTTPHeaders)
+	if _, ok := header["User-Agent"]; !ok {
```

Comment:
I think there's a bug here, since headers are case-insensitive. I think `buildHeader` should use the header's `Add` method, which canonicalizes keys, and then use `header.Get("User-Agent")` here. Note that this is a slight behavior change though, which could affect probers...

Reply:
That won't affect probers, because this code isn't used for probers; it's the deprecated old path specific to lifecycle handlers.